src/spikeinterface/sortingcomponents/motion/dredge.py

```diff
@@ -0,0 +1,1469 @@
+    Given a DxT raster, this computes normalized cross correlations for
+    all pairs of time bins at offsets in the range [-disp, disp], by
+    increments of step_size. Then it finds the best one and its
+    corresponding displacement, resulting in two TxT matrices: one for
+    the normxcorrs at the best displacement, and the matrix of the best
+    displacements.
+
+    Arguments
+    ---------
+    raster : DxT array
+    batch_size : int
+        How many raster rows to xcorr against the whole raster
+        at once.
+    step_size : int
+        Displacement increment. Not implemented yet but easy to do.
+    disp : int
+        Maximum displacement
+    device : torch device
+    Returns: D, C: TxT arrays
+    """
+    import torch
+
+    D, Ta = raster_a.shape
+    D_, Tb = raster_b.shape
+
+    # sensible default: at most half the domain.
+    if disp is None:
+        disp == D // 2
+
+    # range of displacements
+    if D == D_:
+        if possible_displacement is None:
+            possible_displacement = np.arange(-disp, disp + 1)
+    else:
+        assert possible_displacement is not None
+        assert disp is not None
+
+    # pick torch device if unset
+    if device is None:
+        device = (
+            torch.device("cuda")
+            if torch.cuda.is_available()
+            else torch.device("cpu")
+        )
+
+    # process rasters into the tensors we need for conv2ds below
+    # convert to TxD device floats
+    raster_a = torch.as_tensor(raster_a.T, dtype=torch.float32, device=device)
+    # normalize over depth for normalized (uncentered) xcorrs
+    raster_b = torch.as_tensor(raster_b.T, dtype=torch.float32, device=device)
+
+    D = np.zeros((Ta, Tb), dtype=np.float32)
+    C = np.zeros((Ta, Tb), dtype=np.float32)
+    for i in range(0, Ta, batch_size):
+        for j in range(0, Tb, batch_size):
+            dt_bins = min(
+                abs(i - j), abs(i + batch_size - j), abs(i - j - batch_size)
+            )
+            if max_dt_bins and dt_bins > max_dt_bins:
+                continue
+            weights_ = weights
+            if masks is not None:
+                weights_ = masks.T[i : i + batch_size] * weights
+            corr = normxcorr1d(
+                raster_a[i : i + batch_size],
+                raster_b[j : j + batch_size],
+                weights=weights_,
+                xmasks=None if xmasks is None else xmasks.T[j : j + batch_size],
+                padding=disp,
+                normalized=normalized,
+                centered=centered,
+            )
+            max_corr, best_disp_inds = torch.max(corr, dim=2)
+            best_disp = possible_displacement[best_disp_inds.cpu()]
+            D[i : i + batch_size, j : j + batch_size] = best_disp.T
+            C[i : i + batch_size, j : j + batch_size] = max_corr.cpu().T
+
+    return D, C
+
+
+# TODO charlie sam : at the moment this is a duplicate with small differences see motion_estimate.py same function
```

Comment:
I checked through these two functions. There is no functional difference between them, and it should be fine to switch to the new one everywhere.

Reply:
The main difference is this `xmasks` thing which is not used, but I think it's fine to keep it if you don't mind. Or else in my PR I can remove it, but I think let's stick to the new function if that's fine just for other tiny changes.

Reply:
OK.
I remove the one in decentralized